app/src/main/java/com/github/se/eventradar/ui/navigation/NavGraph.kt

```diff
@@ -61,10 +62,12 @@ fun NavGraph(
           val viewModel = ViewFriendsProfileViewModel.create(friendUserId = friendUserId)
           ViewFriendsProfileUi(viewModel = viewModel, navigationActions = navActions)
         }
-
     // TODO replace the Toast message with the corresponding screen function of the route
     composable(Route.MESSAGE) { MessagesScreen(navigationActions = navActions) }
-    composable(Route.SCANNER) { QrCodeScreen(navigationActions = navActions) }
+    composable(Route.SCANNER) {
+      val viewModel = ScanFriendQrViewModel.create(navigationActions = navActions)
+      QrCodeScreen(viewModel = viewModel, navigationActions = navActions)
```

Comment:
There shouldn't be a need to pass navigation actions into both the UI and the VM.